blockbase/inc/fonts/custom-fonts.php

```diff
@@ -205,12 +199,37 @@ function enqueue_block_fonts( $content, $parsed_block ) {
 	return $content;
 }
 
-/**
- * Jetpack may attempt to register fonts for the Google Font Provider.
- * If that happens on a child theme then ONLY Jetpack fonts are registered.
- * This 'filter' filters out all of the fonts Jetpack should register
- * so that we depend exclusively on those provided by Blockbase.
- */
-function blockbase_filter_jetpack_google_fonts_list( $list_to_filter ) {
-	return array();
+function blockbase_register_fonts() {
+	if ( ! function_exists( 'wp_register_webfonts' ) ) {
+		return;
+	}
+
+	$font_families = collect_fonts_from_blockbase();
+	foreach ( $font_families as $font_family ) {
+		wp_register_webfonts(
+			array(
+				array(
+					// 'name' key will eventually be supported: https://github.com/WordPress/gutenberg/issues/46398,
+					// in cases where we want the font's display name to differ from the 'font-family' key.
+					'font-family'  => $font_family['name'],
+					'font-weight'  => '100 900',
+					'font-style'   => 'normal',
+					'font-display' => 'fallback',
+					'provider'     => $font_family['provider'],
+				),
+				array(
+					'font-family'  => $font_family['name'],
+					'font-weight'  => '100 900',
+					'font-style'   => 'italic',
+					'font-display' => 'fallback',
+					'provider'     => $font_family['provider'],
+				),
+			)
```

Comment:
FYI using `wp_register_webfonts` with this array signature is fine _today_, we'll just need to migrate to new format once the new API stabilizes, as this exact format got deprecated.

More in https://github.com/WordPress/gutenberg/pull/43492

Reply:
Gotcha. If this approach is acceptable, we can open a follow-up PR to change the format and merge it after the new API lands in a Gutenberg 14.9.

Reply:
Yep exactly. We'll also need to migrate Jetpack https://github.com/Automattic/jetpack/pull/28054/files

Reply:
More info here: https://github.com/Automattic/jetpack/issues/28063